awx/api/serializers.py

```diff
@@ -817,6 +824,25 @@ def to_representation(self, obj):
         ret['job_explanation'] = _(obj.job_explanation)
         return ret
 
+    def get_launched_by(self,obj):
+        
+        #sdb.set_trace()
+        dictionary ={}
+        if obj.launched_by is not None:
+            dictionary = model_to_dict(obj.launched_by)
+            return dictionary
+        else:
+            return {}
+    
+    def get_ancestor_job(self,obj):
+
+        if obj.ancestor_job is not None:
```

Comment:
Your `ancestor_job` property is written to always return something.  It should be reasonable for this method to just `return model_to_dict(obj.ancestor_job)`

Reply:
The `model_to_dict` does not handle `None` . It throws an error. `NoneType` returns `None` here and it breaks . That is why i did this :
```
if obj.launched_by is not None:
            dictionary = model_to_dict(obj.launched_by)
            return dictionary
        else:
            return {}

```

If `None` goes into model_to_dict , it breaks.